packages/middleware-sdk-ec2/src/index.ts

```diff
@@ -40,17 +40,38 @@ export function copySnapshotPresignedUrlMiddleware(options: PreviouslyResolved):
     ): InitializeHandler<any, Output> =>
     async (args: InitializeHandlerArguments<any>): Promise<InitializeHandlerOutput<Output>> => {
       const { input } = args;
+
       if (!input.PresignedUrl) {
-        const region = await options.region();
-        const resolvedEndpoint =
-          typeof options.endpoint === "function" ? await options.endpoint() : toEndpointV1(context.endpointV2!);
+        const destinationRegion = await options.region();
 
-        if (typeof options.regionInfoProvider === "function") {
-          const regionInfo = await options.regionInfoProvider(input.SourceRegion);
-          resolvedEndpoint.hostname = regionInfo?.hostname || `ec2.${input.SourceRegion}.amazonaws.com`;
-        } else {
-          resolvedEndpoint.hostname = `ec2.${input.SourceRegion}.amazonaws.com`;
```

Comment:
with regionInfoProvider gone as of endpoints v2, all calls defaulted to this else block.

This domain only works for commercial regions, so the fix is to use the endpoints v2 resolver on both the presigned URL with the source region and the base request with the target region.